.github/workflows/cuda-label-remover.yml

```diff
@@ -0,0 +1,23 @@
+name: Remove "CUDA CI" Label
+
+# This workflow removes the "CUDA CI" label that triggers the actual
+# CUDA CI. It is separate so that we can use the `pull_request_target`
+# trigger which has a API token with write access.
+on:
```

Comment:
Will this work all the time?

For example, if the "Remove label" and CUDA GPU workflows were queued at the same time and "Remove label" completes first, then the label is removed and CUDA GPU workflow is skipped.

I think an alternative is to use a `workflow_run` which also has write permissions:

```
on:
  workflow_run:
    # Run the workflow after the separate "Documentation builder" workflow completes
    workflows: [CUDA GPU]
    types: [requested]
```

Reply:
I don't know :-/ It seems like some of the "repo" is frozen in time. For example if you trigger a workflow, have `sleep 30` as its first step, push new commits during that sleep and then use the default `checkout` action you get the commit you were expecting (not the newly pushed one).

The `if` statement in the actual CUAD CI job evaluates against the `event` information (doesn't query the repo in real time), which you would hope is a snapshot taken at the time of the event. But, I don't know how to test that hypothesis. 

---

Having said all that if we can trigger it on the completion (or start?) of another workflow that would be simpler. I'll investigate a bit. I think we want to remove the label independent of whether the tests pass or not, and probably "as soon as reasonable" after the CUDA CI workflow starts?

Reply:
Do you know how to access the PR number in the `workflow_run` triggered workflow? The options I can find rely on writing it to an artefact in the first job, uploading it and then fetching it in the followup workflow.

Reply:
You'd think https://github.com/betatim/scikit-learn-ci/blob/d4f2d0bbdf072b0a3e1bede34205f559f46d08a1/.github/workflows/label-remover.yml would work, but it fails https://github.com/betatim/scikit-learn-ci/actions/runs/9888261056/job/27311767865#step:2:9

Reply:
Using `${{ github.event.workflow_run.pull_requests[0].number }}` would work if the PR wasn't from a fork.

Reply:
Doing a bit of thinking about it, I'd vote for going with what is in the PR for now. If we do end up with races then we will see it, workflows will be marked as "skipped", and can come back to fix it.

I think we won't see races because the `event` information is captured at the time the event is emitted. That is the only sane way to build a system like this, lets hope GitHub engineering agrees with me :D

The alternative seems to involve writing the PR number to an artefact in the `pull_request` workflow, uploading it and fetching it again in this workflow. This seems like a lot of "stuff" to add and makes me wonder if there are hidden traps in sharing information like this (similar to cache poisoning).

Reply:
Okay lets go with this PR's solution.